src/Toolkit/Dom.php

```diff
@@ -655,6 +679,16 @@ protected function exportHtml(bool $normalize = false): string
 		static::remove($metaTag);
 		$html = str_replace($this->doc->saveHTML($metaTag), '', $html);
 
+		// if the original input contained an HTML doctype, some libxml
+		// implementations expand it to the long HTML4 transitional doctype
+		// when saving. Normalize it back to the short `<!DOCTYPE html>`
+		// to keep behavior consistent across environments.
+		if (Str::contains($this->code, '<!DOCTYPE ', true) === true) {
+			if (preg_match('/<!doctype\s+html/i', $this->code) === 1) {
+				$html = preg_replace('/^<!DOCTYPE[^>]*>\s*/i', '<!DOCTYPE html>' . "\n", $html, 1);
+			}
+		}
```

Comment:
`Suggestion`: To reduce nesting:


```suggestion
		if (
			Str::contains($this->code, '<!DOCTYPE ', true) === true &&
			preg_match('/<!doctype\s+html/i', $this->code) === 1
		) {
			$html = preg_replace('/^<!DOCTYPE[^>]*>\s*/i', '<!DOCTYPE html>' . "\n", $html, 1);
		}
```